measures/qoi_report/measure.rb

```diff
@@ -456,6 +497,35 @@ def daily_peak_timing_stats_by_month(timeseries, month_val, stats_option = 'med'
     stats_by_month
   end
 
+  def grid_peak_schedule(model, peak_len)
+    grid_region = model.getBuilding.additionalProperties.getFeatureAsString('grid_region')
+    raise 'Unable to find grid region in model building additional properties' unless grid_region.is_initialized
+    grid_region = grid_region.get
+    load_csv = "#{File.dirname(__FILE__)}/cambium/#{scenario}/#{grid_region}.csv"
+    return nil if !File.file?(load_csv)
+    net_load_mwh = CSV.read(load_csv, converters: :float).flatten
+    grid_peak_schedule = peak_schedule_generation(annual_load = net_load_mwh, oat = timeseries['temperature'], peak_len = peak_len, num_timesteps_in_hr = 1, peak_window_strategy = 'center with peak', rebound_len = 0, prepeak_len = 0, season = 'all')
+    grid_peak_schedule
+  end
+
+  def daily_peak_stats_on_grid_peak_by_month(model, peak_len, timeseries, month_val, stats_option = 'mean', year = 2018)
+    grid_peak_schedule = grid_peak_schedule(model, peak_len)
+    daily_peaks = timeseries['total_site_electricity_kw'].zip(grid_peak_schedule).map { |kw, peak| kw * peak }
+    daily_peak_by_month = []
+    daily_peaks.each_slice(24).with_index do |kws, doy|
+      month, = day_of_year_to_date(doy, year)
+      daily_peak_by_month << (kws.sum(0.0) / peak_len) if month == month_val
+    end
+    stats_by_month = nil
+    case stats_option
+    when 'mean'
+      stats_by_month = daily_peak_by_month.sum(0.0) / daily_peak_by_month.size
+    else
+      return nil
```

Comment:
hm what about returning an error so we catch if anything happens other than 'mean'?

Reply:
Yes although the usage of this function is hardcoded with 'mean' input within this measure.